test/functional/tools/job_environment_default.xml

```diff
@@ -12,6 +12,8 @@ echo \$(pwd) > '$pwd' &&
 echo "\$HOME" > '$home' &&
 echo "\$TMP"  > '$tmp' &&
 echo "\$SOME_ENV_VAR" > '$some_env_var' &&
+echo "\${JOBCONF_ENV_VAR:-UNSET}" > '$jobconf_env_var' &&
```

Comment:
Can you remove this ? We should fix this, not cement it with tests.

Reply:
Can we leave this: It just tests the documented behavior (as per this PR :) ) and you gave good reasons for not changing it. 

If at one time we allow certain env variables to be passed to the container, e.g. via `ToolInfo` then we can easily add another variable for this behavior.